include/onnxruntime/core/providers/openvino/openvino_provider_factory.h

```diff
@@ -5,13 +5,16 @@
 
 #ifdef __cplusplus
 extern "C" {
+#else
+#include <stdbool.h>
 #endif
 
 /**
  * \param device_id openvino device id, starts from zero.
  */
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_OpenVINO,
-    _In_ OrtSessionOptions* options, const char* device_id);
+    _In_ OrtSessionOptions* options, const char* device_type,
```

Comment:
This will break the ABI. You'll need to add one more function with new params. I suggest adding a flexible way to pass a bag of key/val pairs to the EP so that you don't have to change the API every single time.

Reply:
> This will break the ABI. You'll need to add one more function with new params. I suggest adding a flexible way to pass a bag of key/val pairs to the EP so that you don't have to change the API every single time.

should the EP have its own param of key/value pairs, or can the EP use session configuration options in SessionOptions that was introduced in https://github.com/microsoft/onnxruntime/pull/4718/

Reply:
You could choose any approach. However, with session configs, you'll have to add an entry to include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h. This file gets included by the main C API header. I'm not sure if it's a good idea for configs from different EPs to pollute this file even when the user is not using the EP. EP configs should be localized to EPs only.

Reply:
in that case, probably better to add a new function with new params (one that uses key/value pairs for options) as you originally suggested.

Reply:
Created a second C/C++ API named OrtSessionOptionsAppendExecutionProvider_OpenVINO**EP** that takes key-value pair settings all appended as a single string with delimiters.